src/edu/washington/escience/myria/api/encoding/QueryConstruct.java

```diff
@@ -85,18 +85,21 @@
       }
     }
 
-    Map<Integer, SubQueryPlan> plan = Maps.newHashMap();
-    Map<PlanFragmentEncoding, RootOperator> instantiatedFragments = Maps.newHashMap();
     Map<Integer, Operator> allOperators = Maps.newHashMap();
     for (PlanFragmentEncoding fragment : fragments) {
-      RootOperator op =
-          instantiateFragment(fragment, args, instantiatedFragments, op2OwnerFragmentMapping, allOperators);
+      instantiateFragmentOperators(fragment, args, allOperators);
+    }
+    while (setConsumerSchema(fragments, allOperators)) {
```

Comment:
I'm still concerned about infinite loop bugs. Could we set an arbitrary "big enough" iteration limit and either assert on it or log an error and throw an exception? It's not elegant but could save a lot of debugging.